src/webui/abstractwebapplication.h

```diff
@@ -105,7 +105,7 @@ private slots:
     bool sessionInitialize();
 
     QStringMap parseCookie(const Http::Request &request) const;
-    bool isCrossSiteRequest(const Http::Request &request) const;
+    bool isCrossSiteRequest(const Http::Request &request, const Http::Environment &env) const;
```

Comment:
Didn't pay attention before...
These two methods have incorrect design. There is no need to pass class members as parameters of non-static class methods. You can use request_ and env_ inside it directly.

Reply:
You missed my ps. in the first post.
As you can see `isCrossSiteRequest()` depends solely on `Http::Request ` members and the addition of `Http::Environment/QHostAdress` is purely for debug message use.
I plan to move this method under `Http::Request` class, thus I still need `Http::Environment` or `QHostAdress` in the future.


BTW, I see no reason why there are 2 separate classes: `Http::Request` `Http::Environment`, IMO they should be merged together, but that is for later.

Reply:
>BTW, I see no reason why there are 2 separate classes:

The main reason is Environment contains data that is not part of request itself (client address is not even part of HTTP protocol).

Reply:
>The main reason is Environment contains data that is not part of request itself (client address is not even part of HTTP protocol).

I understand your point now, yet I think for coding convenience, there should be a super set containing the 2 structs (just an idea)

---

>You missed my ps. in the first post.

What do you think? or I should use class members now and refactor later?

Reply:
> or I should use class members now and refactor later?

Yes.

Reply:
>You can use request_ and env_ inside it directly.

OK, done.

Reply:
But you still pass request as parameter...

Reply:
If you don't want to fix existing code then ok, will do it later.

Reply:
I would like to minimize the changes now.